flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/DefaultSlotPoolServiceSchedulerFactoryTest.java

```diff
@@ -74,6 +76,69 @@ public void testAdaptiveSchedulerForReactiveMode() {
                 .isEqualTo(JobManagerOptions.SchedulerType.Adaptive);
     }
 
+    @Test
+    public void testFallBackSchedulerWithAdaptiveTestProperty() {
+        Properties properties = System.getProperties();
+        boolean containsAdaptive = properties.containsKey("flink.tests.enable-adaptive-scheduler");
+        if (!containsAdaptive) {
+            System.getProperties().setProperty("flink.tests.enable-adaptive-scheduler", "true");
+        }
```

Comment:
It would be better to introduce two methods,
```
private Optional<Tuple2<String, String>> saveAdaptiveSchedulerTestProperty();
private void restoreAdaptiveSchedulerTestProperties(Optional<Tuple2<String, String>> savedProperty);
```

And always do save at the beginning and do restore in the end.